### ISSUE TITLE
[AQ40] Twin Emperors - Teleport

### ISSUE THREAD

**Issue by Si1ker** [chromiecraft/chromiecraft#4264]
### What client do you play on?

enUS

### Faction

Both

### Content Phase:

60

### Current Behaviour

Vek’lor and Vek’nilash start their attacks right after swapping, sometimes this makes one of the two bosses, Vek’nilash for example, go after its previous target to try to cast a new shadowbolt.

### Expected Blizzlike Behaviour

Vek’lor and Vek’nilash should have a delay of 2 seconds before retaking their attacks after the swamp. They complete the full teleport spell animation, remove anyone targeted, stop all actions and then retake their normal cycle of attacks.

### Source

explanation: https://youtu.be/ulzbebfPxcg?t=261

example 1: https://youtu.be/ulzbebfPxcg?t=194
example 2: https://youtu.be/ulzbebfPxcg?t=233
example 3: https://youtu.be/ulzbebfPxcg?t=298
example 4: https://youtu.be/ulzbebfPxcg?t=330

### Steps to reproduce the problem

Go AQ40
Engage combat in this encounter

### Extra Notes

Without this period of grace, the fight gets buggy with threat and more challenging than it was designed to be as that short delay allows for the other tank/warlock to take initial action.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/b427e8e18cb6baa80124ce8e3a7e79aa670bcc74

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Fayz2** [comment]
To add to this

After teleport the Veknilash/Veklor tank closest will have 2k threat.  This is why in the videos you link above the caster mob often arcane explosions the physical tank and shadowbolts him until the warlock is able to rip him off with searing pain.

**Comment by Si1ker** [comment]
> 

Yep, in a blizzlike behaviour that's what you would expect which us ok.
On Live, the caster twin is trying to go for the PREVIOUS target like if it added threat to the previous target it had before the TP
